src/rust-crypto/rust-crypto.ts

```diff
@@ -151,7 +153,24 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
      * Implementation of {@link CryptoBackend#getBackupDecryptor}.
      */
     public async getBackupDecryptor(backupInfo: KeyBackupInfo, privKey: ArrayLike<number>): Promise<BackupDecryptor> {
-        throw new Error("Stub not yet implemented");
+        if (backupInfo.algorithm != "m.megolm_backup.v1.curve25519-aes-sha2") {
+            throw new Error(`getBackupDecryptor Unsupported algorithm ${backupInfo.algorithm}`);
+        }
+
+        const authData = <Curve25519AuthData>backupInfo.auth_data;
+
+        if (!(privKey instanceof Uint8Array)) {
+            // eslint-disable-next-line @typescript-eslint/no-base-to-string
+            throw new Error(`getBackupDecryptor expects Uint8Array, got ${privKey}`);
+        }
+
+        const backupDecryptionKey = RustSdkCryptoJs.BackupDecryptionKey.fromBase64(encodeBase64(privKey));
```

Comment:
`olmlib` looks and sounds like something libolm-related, even if it's not.

`encodeBase64` is pretty simple and I prefer just to inline it:

```suggestion
        const b64PrivKey = Buffer.from(privKey).toString("base64");
        const backupDecryptionKey = RustSdkCryptoJs.BackupDecryptionKey.fromBase64(b64PrivKey);
```

Or, since we already have exactly this code in `storeSessionBackupPrivateKey`, maybe we should factor out a new utility function

Reply:
Done. I created a new base64.ts in CommonCrypto. I updated rust-crypto.ts to use it. 
I haven't touch legacy code that is still using the one from olmlib.